CMakeLists.txt

```diff
@@ -1,332 +1,335 @@
-cmake_minimum_required(VERSION 3.14)
+# Ensure CMake policies have defaults depending on the CMake version used
+# between the two versions specified. e.g. if 3.18 is used, then 3.18 defaults
+# will be used instead of 3.17 defaults.
+cmake_minimum_required(VERSION 3.17...3.26)
 
+# Policies ====================================================================
 
 # Use `LOCATION` for python lookup strategy
 # https://cmake.org/cmake/help/latest/policy/CMP0094.html
-if(POLICY CMP0094)
-    cmake_policy(SET CMP0094 NEW)
-endif()
+cmake_policy(SET CMP0094 NEW)
+
+# find_package() uses <PackageName>_ROOT variables.
+# https://cmake.org/cmake/help/latest/policy/CMP0074.html
+cmake_policy(SET CMP0074 NEW)
 
-if(MPI_ACTIVE)
-   set (CMAKE_CXX_COMPILER "mpicxx")
+# find_package() uses upper-case <PACKAGENAME>_ROOT variables.
+# https://cmake.org/cmake/help/latest/policy/CMP0144.html
+if(POLICY CMP0144)
+    cmake_policy(SET CMP0144 NEW)
```

Comment:
For other reviewers... that is, it uses them _also_... So this makes either `NetCDF_ROOT` _or_ `NETCDF_ROOT` work.